File: apps/python/lib/application/fizz_buzz_value_command.py
from lib.application.fizz_buzz_command import FizzBuzzCommand
from lib.domain.model.fizz_buzz_value import FizzBuzzValue
from lib.domain.type.fizz_buzz_type import FizzBuzzType


class FizzBuzzValueCommand(FizzBuzzCommand):
    """単一値の FizzBuzz を実行するコマンド。"""

    def __init__(self, type_: FizzBuzzType, number: int) -> None:
        self._type = type_
        self._number = number

    def execute(self) -> FizzBuzzValue:
        return self._type.generate(self._number)

File: apps/php/src/Application/FizzBuzzValueCommand.php
<?php

declare(strict_types=1);

namespace App\Application;

use App\Domain\Model\FizzBuzzValue;
use App\Domain\Type\FizzBuzzType;

final class FizzBuzzValueCommand implements FizzBuzzCommand
{
    public function __construct(
        private readonly FizzBuzzType $type,
    ) {
    }

    public function execute(int $number = 0): FizzBuzzValue
    {
        return $this->type->generate($number);
    }
}
